WordPress/Classes/ViewRelated/Me/MeViewController.swift

```diff
@@ -75,9 +82,32 @@ class MeViewController: UITableViewController, UIViewControllerRestoration {
         reloadViewModel()
     }
 
-    @objc private func accountDidChange() {
+    func switchAccountHelper() -> AccountSelectionHelper {
+        var navBarHeight = self.navigationController?.navigationBar.frame.size.height
+        if (navBarHeight == nil) {
+            navBarHeight = 44
+        }
+        let accountHelper = AccountSelectionHelper(parentView: self.view,
+                                                   accounts: [],
+                                                   delegate: self,
+                                                   height: navBarHeight!)
+
+        accountHelper.delegate = self
+
+        self.retrieveAccounts { (accounts: [Account]) in
+            self.accountHelper?.accounts = accounts
+        }
```

Comment:
As I mentioned in another comment, I don't think there's going to be accounts without emails, and so the retrieve method won't actually perform a network request and it would return immediately. But if it did, as I understand this, you would be able to tap on the account switcher and get an empty list, because this is waiting to load some account details and the helper `accounts` array is empty by default. Moreover, when the details finally arrived, I believe the switcher wouldn't update itself until closed an opened again.

I think it might be better to not set this manually and have the helper use a `NSFetchedResultsController` so it gets automatic updates. AFAIK, you're only ensuring we have an email so we can show a gravatar, and that can definitely be asynchronous (even if you have the email, the image already loads asynchronously).